src/syntax/parser.rs

```diff
@@ -538,6 +544,9 @@ where
                 Ok(ast::Expression::VariantExpression { id, key: key })
             }
             Some('(') => {
+                if id.name.starts_with('-') {
+                    return error!(ErrorKind::ForbiddenCaller);
```

Comment:
The `ForbiddenCaller` check needs a bit of work, likely outside of this PR. The EBNF requires the callee to be [all uppercase](https://github.com/projectfluent/fluent/blob/06d20408febd5c50da1c6b6e173568b403c07ba5/spec/fluent.ebnf#L30).